### PR TITLE
Skeletonize the project

### LINKED ISSUE #215: Skeletonize repository and standardize code formatting
# 💡 Summary #

This repository still uses Travis instead of GitHub Actions, is not [skeletonized (per our project setup guide)](https://github.com/cisagov/development-guide/tree/develop/project_setup), and the python code is not formatted with `black`, our standard code-formatting tool. 

## Motivation and context ##

Standardize code and repository setup to keep it updated and enable standard linting.

## Acceptance criteria ##

How do we know when this work is done?

- [ ] Repository has been skeletonized (likely `skeleton-python-library`)
- [ ] `black` has reformatted Python files
- [ ] Linters and tests pass

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull requests integrates the [cisagov/skeleton-python-library](https://github.com/cisagov/skeleton-python-library) skeleton as the upstream parent. As part of the process the code base was modified to pass our standard [pre-commit](https://pre-commit.com) configuration and modernized. This closes #215.

## 💭 Motivation and context ##

This project needs to be dragged into our modern development methodology to allow for improvements and maintenance tasks. The most important aspect of this is the inclusion of our GitHub Actions workflow for linting and testing.

## 🧪 Testing ##

Automated tests pass. I ran the existing version on the `develop` branch and this branch against the [CSV of current federal domains](https://github.com/cisagov/dotgov-data/blob/main/current-federal.csv) and skimming a diff of the resultant `results.csv` files only showed differences in resolved IP addresses (likely because of load balancing).

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Bump version.

## ✅ Post-merge checklist ##

- [ ] Add a tag or create a release.

### CODE REVIEW

File: src/pshtt/pshtt.py [hunk 1]
```diff
@@ -793,24 +906,23 @@ def canonical_endpoint(http, httpwww, https, httpswww):
...
     def root_unused(endpoint):
         return (
-            endpoint.redirect or
-            (not endpoint.live) or
-            endpoint.https_bad_hostname or  # harmless for http endpoints
-            (not str(endpoint.status).startswith("2"))
+            endpoint.redirect
+            or not endpoint.live
+            or endpoint.https_bad_hostname
+            or not str(endpoint.status).startswith("2")  # harmless for http endpoints
```
Comment (jsf9k) [inline src/pshtt/pshtt.py]:
```suggestion
            or endpoint.https_bad_hostname  # harmless for http endpoints
            or not str(endpoint.status).startswith("2")
```

**Review by jsf9k — APPROVED** [review]
I had one comment, but this PR is otherwise approved.  Thanks for cleaning these Augean stables, Hercules!

File: gce-scripts/README.md [hunk 2]
```diff
@@ -15,189 +15,183 @@ that.
...
-            *   www.subdomain.domain.tld
-    *   `mkdir ~/pshtt_run/data_results/`
-    *   `mv ~/pshtt_run/combine_shards.py ~/pshtt_run/data_results`
-        *   Places combine_shards.py into data_results/.
-    *   `mkdir ~/pshtt_run/input_files/`
-
-4.  roots.pem
+1. Download the gcloud command line tool.
+
+    - follow the [download
```
Comment (dav3r) [inline gce-scripts/README.md]: Capitalization:
```suggestion
    - Follow the [download
```

File: gce-scripts/README.md [hunk 3]
```diff
@@ -15,189 +15,183 @@ that.
...
+    - If you already have this installed, following the following
+      instructions:
+    - `gcloud init`
+      - Click `2` create a new configuration.
+      - Enter `pshtt-configuration`
+      - Choose the appropriate account
+      - Click the appopriate number corresponding to your google project
+      - If it complains that the API is not enabled, hit enabled and retry.
+      - Do not set default zone or region
+      - at this point, your default project should be this google project.
```
Comment (dav3r) [inline gce-scripts/README.md]: Capitalization:
```suggestion
      - At this point, your default project should be this google project.
```

File: gce-scripts/README.md [hunk 4]
```diff
@@ -15,189 +15,183 @@ that.
...
+
+1. Copy all .sh scripts from this directory:
+
+    - Keep the name of the scripts the same.
+    - `chmod +x ~/pshtt_run/*.sh`
+      - which will make all the scripts executable.
+    - `touch domains.csv`
+      - Your domain list, one domain per line, with the input list ending in
+        `.csv`.
+      - domains must have the schema stripped of them and no trailing '/',
```
Comment (dav3r) [inline gce-scripts/README.md]: Capitalization:
```suggestion
      - Domains must have the schema stripped of them and no trailing '/',
```

File: gce-scripts/README.md [hunk 5]
```diff
@@ -15,189 +15,183 @@ that.
...
-    *   If your new ssh key is called "gce_pshtt_key", skip this step.
-    *   If you did not name your ***new*** ssh key gce_pshtt_key, then you will
-        need to go through and rename the gce_pshtt_key in all the .sh files to
-        whatever you named your key.
-    *   in vim, this is ":%s/gce_pshtt_key/yourkeynamehere/g <enter>".
+    - If your new ssh key is called "gce_pshtt_key", skip this step.
+    - If you did not name your ***new*** ssh key gce_pshtt_key, then you will
+      need to go through and rename the gce_pshtt_key in all the .sh files to
+      whatever you named your key.
+    - in vim, this is `:%s/gce_pshtt_key/yourkeynamehere/g <enter>`.
```
Comment (dav3r) [inline gce-scripts/README.md]: Capitalization:
```suggestion
    - In vim, this is `:%s/gce_pshtt_key/yourkeynamehere/g <enter>`.
```

File: gce-scripts/README.md [hunk 6]
```diff
@@ -15,189 +15,183 @@ that.
...
+    - each machine will contain a shard of the data to run.
+    - This is the script that sets up all machines and puts all datafiles on
+      the machines for running.
+    - `./run_all_scripts top-1m.nocommas.8.31.2017 100 alexa`
+    - will produce 100 shards all starting with "alexa" in the input_files
+      dir.
+      - ex. alexa000.csv
+    - NOTE: you can ONLY create 999 shards. If you need more than 999 shards,
+      you will need to change the split_up_dataset.sh file.
+1. exit screen `cntr+a+d`
```
Comment (dav3r) [inline gce-scripts/README.md]: Capitalization:
```suggestion
    - Number of shards == number of hosts
    - Each machine will contain a shard of the data to run.
    - This is the script that sets up all machines and puts all datafiles on
      the machines for running.
    - `./run_all_scripts top-1m.nocommas.8.31.2017 100 alexa`
    - Will produce 100 shards all starting with "alexa" in the input_files
      dir.
      - ex. alexa000.csv
    - NOTE: you can ONLY create 999 shards. If you need more than 999 shards,
      you will need to change the split_up_dataset.sh file.
1. Exit screen `cntr+a+d`
```

File: gce-scripts/README.md [hunk 7]
```diff
@@ -15,189 +15,183 @@ that.
...
+
+### During the run ###
+
+- `./check_instances.sh`
+  - will print the ip of each host, as well as FINISHED or NOT FINISHED.
+
+### After the run ###
+
+- `./grab_and_combine_data.sh`
+  - will grab all log and result data files, combine data files into one
```
Comment (dav3r) [inline gce-scripts/README.md]: Capitalization:
```suggestion
  - Will print the ip of each host, as well as FINISHED or NOT FINISHED.

### After the run ###

- `./grab_and_combine_data.sh`
  - Will grab all log and result data files, combine data files into one
```

File: gce-scripts/README.md [hunk 8]
```diff
@@ -15,189 +15,183 @@ that.
...
-    *   If you want to be able to use this json file with any of the colab
-        notebooks (like the one listed below), you will also need to run
-        combine_shards.py.into the same dir as the json file.
-        *   Copy combine_shards.py into the same dir as the json file.
-        *   `echo <input_filename>.json > to_combine.txt`
-        *   `python combine_shards.py to_combine.txt > final_results.json`
-    *   Log can be found in `time_<input_filename>.txt`.
+    - Follow directions under Updating data files above on how to get a
+      roots.pem.
+    - domains must have the schema stripped of them and no trailing '/', such
```
Comment (dav3r) [inline gce-scripts/README.md]: Capitalization:
```suggestion
    - Domains must have the schema stripped of them and no trailing '/', such
```

File: setup.py [hunk 9]
```diff
@@ -1,94 +1,125 @@
...
+        "wget>=3.2",
+        "docopt>=0.6.2",
+        "pytablereader>=0.15.0",
+        "pytablewriter>=0.27.2",
+        "publicsuffix>=1.1.0",
+        "pyopenssl>=17.5.0",
+        "python-dateutil>=2.7.3",
+        "pytz>=2018.5",
+        # This is necessary to support the python_requires kwarg
+        "setuptools >= 24.2.0",
```
Comment (dav3r) [inline setup.py]: Alphabetize:
```suggestion
        "docopt>=0.6.2",
        "publicsuffix>=1.1.0",
        "pyopenssl>=17.5.0",
        "pytablereader>=0.15.0",
        "pytablewriter>=0.27.2",
        "python-dateutil>=2.7.3",
        "pytz>=2018.5",
        "requests>=2.18.4",
        # This is necessary to support the python_requires kwarg
        "setuptools >= 24.2.0",
        "sslyze>=2.1.3,<3.0.0",
        "wget>=3.2",
```

**Review by dav3r — APPROVED** [review]
Great work Hercules!  I mentioned some small thangs that could be cleaned up along the way here, but thanks for taking on this massive effort!
image

**Review by dav3r — APPROVED** [review]
Since @jsf9k isn't around at the moment, I get to steal his line....

**_Approval intensifies!_**

**Review by jsf9k — APPROVED** [review]
Strong work!

**Pull request merged**